### PR TITLE
Implement SendBit functionality and wrap into Stream API

### LINKED ISSUE #1: Name and reference link
It will probably be better to call the library only Arduino Sigfox, otherwise if we will release in the future other products that may use this library it could get confusing.

The correct reference link for the description is https://www.arduino.cc/en/Reference/SigFox

### PR DESCRIPTION
**Pull request by facchinm**
This PR incorporates and superseded #2 .
FIxes #1 

@nicolsc

### CODE REVIEW

**Comment by nicolsc** [comment]
Thx @facchinm  !

File: src/SigFox.cpp [hunk 1]
```diff
@@ -122,6 +122,12 @@ int SIGFOXClass::begin(SPIClass& spi, int reset, int poweron, int interrupt, int
 int SIGFOXClass::send(unsigned char mess[], int len, bool rx)
 {
   if (len == 0) return -1;
+
+  if (rx == false && len == 1 && mess[0] < 2) {
+    //we can use send_bit command
```
Comment (sandeepmistry) [inline src/SigFox.cpp]: @nicolsc @facchinm will this change behaviour if the remote side is expecting a 0 or 1 in a single byte?

Comment (facchinm) [inline src/SigFox.cpp]: As far as I understand, on the backend, the received value becomes an `uint8_t`, it is only sent as a single bit at the physical layer.

Comment (nicolsc) [inline src/SigFox.cpp]: :+1: @facchinm 

It's purely protocol-related .

On the backend/callback side, it's handled as an standard payload & exposed as an hexadecimal string to convert back to original data types.

**Pull request merged**